ext/enchant/enchant.c

```diff
@@ -626,7 +626,7 @@ PHP_FUNCTION(enchant_dict_quick_check)
 	size_t wordlen;
 	enchant_dict *pdict;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "rs|z", &dict, &word, &wordlen, &sugg) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "rs|z!", &dict, &word, &wordlen, &sugg) == FAILURE) {
```

Comment:
This doesn't really make sense to me. `sugg` is a by-reference parameter, so it will usually be null to start with, and will not get populated.

It should be fine to simply specify the `null` default, without doing any code changes. This is more about optimization (don't populate the parameter if it's not passed) than about behavior.

Reply:
My understanding was that unless `z!` is specified, `null` won't be possible to be passed. But it wouldn't really make sense, I agree.

Reply:
`z` does not perform any type checks. In this case `!` only determined whether you get back a zval (with a null value) or a null pointer.

Reply:
Ah yes! I see the distinction now (and I should already have seen it)